kroxylicious-api/src/test/java/io/kroxylicious/proxy/service/BaseContributorTest.java

```diff
@@ -164,4 +222,79 @@ void shouldThrowForUnknownTypeNameConfigDefinition() {
 
         // Then
     }
+
+    @Test
+    void shouldFailIfConfigIsNullAndRequired() {
+        // Given
+        BaseContributor.BaseContributorBuilder<Long, Context> builder = BaseContributor.builder();
+        builder.add("requiresConfig", LongConfig.class, (baseConfig, context) -> 1L, true);
+        BaseContributor<Long, Context> baseContributor = new BaseContributor<>(builder) {
+        };
+
+        // When
+        assertThatThrownBy(() -> baseContributor.getInstance("requiresConfig", () -> null)).isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("'requiresConfig' requires configuration but none was supplied");
+
+        // Then
```

Comment:
these dangling thens look a bit odd

Reply:
Hmm, yeah. 

Just because I can doesn't mean I should combine `when` and `then`